content/source/docs/extend/best-practices/versioning.html.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: "extend"
+page_title: "Extending Terraform: Versioning Best Practices"
+sidebar_current: "docs-extend-best-practices-versioning"
+description: |-
+  Recommendations for version numbering and documentation.
+---
+
+# Versioning
+
+Given the breadth of available Terraform providers, ensuring a consistent experience means setting a standard guideline for compatibility promises. These guidelines are enforced for providers released by HashiCorp and are recommended for all community providers.
+
+## Table of Contents
+
+- [Versioning Specification](#versioning-specification)
+- [Changelog Specification](#changelog-specification)
+  - [Version Headers](#version-headers)
+  - [Catagorization](#catagorization)
+  - [Entry Format](#entry-format)
+  - [Entry Ordering](#entry-ordering)
+  - [Example Changelog](#example-changelog)
+
+## Versioning Specification
+
+Observing that Terraform providers are in many ways analogous to shared libraries in a programming language, we adopted a version numbering scheme for providers that follows the guidelines of [Semantic Versioning](http://semver.org/). In summary, this means that with a version number of the form `MAJOR`.`MINOR`.`PATCH`, the following meanings apply:
+
+- Increasing only the patch number suggests that the release includes only bug fixes, and is intended to be functionally equivalent.
+- Increasing the minor number suggests that new features have been added but that existing functionality remains broadly compatible.
+- Increasing the major number indicates that significant breaking changes have been made, and thus extra care or attention is required during an upgrade.
+
+Version numbers above `1.0.0` signify stronger compatibility guarantees, based on the rules above.
+
+## Changelog Specification
+
+For better operator experience, we provide a standardized format so development information is available across all providers consistently. The changelog should live in a top level file in the project, named `CHANGELOG` or `CHANGELOG.md`. We generally recommend that the changelog is updated outside of pull requests unless a clear process is setup for handling merge conflicts.
+
+### Version Headers
+
+The upcoming release version number is always at the top of the file and is marked specifically as `(Unreleased)`, with other previously released versions below.
+
+~> **NOTE:** For HashiCorp released providers, the release process will replace the "Unreleased" header with the current date. This line must be present with the target release version to successfully release that version.
+
+```md
+## X.Y.Z (Unreleased)
```

Comment:
Interesting! It seems to treat it as a comment. The upcoming commit changes all these to use `text` instead of `md`. We don't care about them being formatted Markdown. 😉